packages/components/src/components/multi-select/multi-select/multi-select-utils.ts

```diff
@@ -0,0 +1,166 @@
+  SelectDropdownDirectionInternal,
+} from '../../../utils/select/select-dropdown';
+import { Theme } from '../../../utils';
+import { forceUpdate } from '@stencil/core';
+import { MultiSelectOptionInternalHTMLProps } from '../multi-select-option/multi-select-option-utils';
+
+export type MultiSelectState = FormState;
+export type MultiSelectDropdownDirection = SelectDropdownDirection;
+
+export const syncMultiSelectOptionProps = (options: HTMLPMultiSelectOptionElement[], theme: Theme): void => {
+  options.forEach((item: HTMLElement & MultiSelectOptionInternalHTMLProps) => {
+    item.theme = theme;
+    forceUpdate(item);
+  });
+};
+
+export const syncNativeSelect = (
+  nativeSelect: HTMLSelectElement,
+  host: HTMLElement,
+  name: string,
+  disabled: boolean,
+  required: boolean
+): void => {
+  nativeSelect.multiple = true;
+  nativeSelect.name = name;
+  nativeSelect.disabled = disabled;
+  nativeSelect.required = required;
+  nativeSelect.ariaHidden = 'true';
+  nativeSelect.tabIndex = -1;
+  nativeSelect.slot = 'select';
+  if (!host.querySelector('SELECT')) {
+    host.prepend(nativeSelect);
+  }
+};
+
+export const updateNativeSelectOptions = (
+  nativeSelect: HTMLSelectElement,
+  multiSelectOptions: HTMLPMultiSelectOptionElement[]
+): void => {
+  nativeSelect.innerHTML = getSelectedOptions(multiSelectOptions)
+    .map((option) => `<option value="${option.value}" selected="${option.selected}">${option.textContent}</option>`)
+    .join('');
+};
+
+export const updateMultiSelectOptionsFilterState = (
+  searchString: string,
+  options: HTMLPMultiSelectOptionElement[]
+): void => {
+  options.forEach((option) => (option.hidden = !optionIncludesSearchString(option, searchString)));
+};
+
+export const optionIncludesSearchString = (option: HTMLPMultiSelectOptionElement, searchString: string): boolean =>
+  option.textContent.toLowerCase().includes(searchString.toLowerCase());
+
+export const hasFilterOptionResults = (options: HTMLPMultiSelectOptionElement[]): boolean =>
+  options?.some((option) => !option.hidden);
+
+export const resetFilteredOptions = (options: HTMLPMultiSelectOptionElement[]): void =>
+  options.forEach((option) => (option.hidden = false));
+
+export const getSelectedOptions = (options: HTMLPMultiSelectOptionElement[]): HTMLPMultiSelectOptionElement[] =>
+  options.filter((option) => option.selected);
+
+export const getSelectedOptionsString = (options: HTMLPMultiSelectOptionElement[]): string =>
+  getSelectedOptions(options)
+    .map((option) => option.textContent)
+    .join(', ');
+
+export const getValidSelectOptions = (options: HTMLPMultiSelectOptionElement[]): HTMLPMultiSelectOptionElement[] =>
+  options.filter((option) => !option.hidden && !option.disabled);
+
+export const getHighlightedOption = (options: HTMLPMultiSelectOptionElement[]): HTMLPMultiSelectOptionElement =>
+  options.find((option) => option.shadowRoot.querySelector('.option').classList.contains('option--highlighted'));
```

Comment:
Why do we read from the previously rendered option? Wouldn't accessing the instance and reading the prop shorter and better?